pkg/debug/checkups/quarantine.go

```diff
@@ -0,0 +1,191 @@
+
+	var (
+		quarantinePathDepths = map[string]int{
+			`C:\Windows\System32\Drivers`: 3,
+			`C:\ProgramData`:              3,
+
+			`/Library/Application Support`: 4,
+		}
+
+		meddlesomeProcessPatterns = []string{
+			`crowdstrike`,
+			`opswat`,
+			`defend`,
+			`defense`,
+			`threat`,
+			`virus`,
+			`quarantine`,
+			`snitch`,
+		}
+	)
+
+	fmt.Fprint(extraFh, "starting quarantine check\n")
+	q.logMeddlesomeProccesses(ctx, extraFh, meddlesomeProcessPatterns)
+	fmt.Fprintf(extraFh, "\nsearching for quarantined files:\n")
+
+	for path, maxDepth := range quarantinePathDepths {
+		fileInfo, err := os.Stat(path)
+		if err != nil {
+			fmt.Fprintf(extraFh, "%s does not exist\n", path)
+			continue
+		}
+
+		if !fileInfo.IsDir() {
+			fmt.Fprintf(extraFh, "expected %s to be a directory, but was not\n", path)
+			continue
+		}
+
+		q.checkDirs(extraFh, 0, maxDepth, path, "quarantine")
+	}
+
+	fmt.Fprintf(extraFh, "total directories checked: %d\n", q.dirsChecked)
+
+	if len(q.quarantineCounts) == 0 {
+		q.status = Passing
+		q.summary = "no quarantine directories found"
+		fmt.Fprint(extraFh, "no quarantine directories found\n")
+		return nil
+	}
+
+	fmt.Fprintf(extraFh, "quarantine directory paths and file counts:\n")
+
+	totalQuarantinedFiles := 0
+
+	for path, count := range q.quarantineCounts {
+		fmt.Fprintf(extraFh, "%s: %d\n", path, count)
+		totalQuarantinedFiles += count
+	}
+
+	if totalQuarantinedFiles == 0 {
+		q.status = Passing
+		q.summary = "no files found in quarantine directories"
+		return nil
+	}
+
+	q.status = Failing
+	q.summary = fmt.Sprintf("found %d quarantined files", totalQuarantinedFiles)
+	return nil
+}
+
+// Recursively scans dir to given max depth. Creates entry for each dir whose path contains the directoryKeyword.
+// Increments quarantine.quarantineCounts for each file found in folder and descendant folders.
+func (q *quarantine) checkDirs(extraFh io.Writer, currentDepth, maxDepth int, dirPath, directoryKeyword string) {
+	if currentDepth > maxDepth {
+		return
+	}
+
+	q.dirsChecked++
+
+	dirNameContainsKeyword := strings.Contains(strings.ToLower(dirPath), directoryKeyword)
+
+	// add entry for each dir that contains the keyword
+	if dirNameContainsKeyword {
+		// create map entry if not exists
+		if _, ok := q.quarantineCounts[dirPath]; !ok {
+			q.quarantineCounts[dirPath] = 0
+		}
+	}
+
+	dirEntries, err := os.ReadDir(dirPath)
+	if err != nil {
+		// some dirs, such as /Library/Application Support/com.apple.TCC can't be read even with sudo
+		// have to give terminal FDA?
+		// so just move on instead of failing
+		fmt.Fprintf(extraFh, "failed to read %s: %s\n", dirPath, err)
+		return
+	}
+
+	for _, dirEntry := range dirEntries {
+		if dirEntry.IsDir() {
+			q.checkDirs(extraFh, currentDepth+1, maxDepth, filepath.Join(dirPath, dirEntry.Name()), directoryKeyword)
+			continue
+		}
+
+		if !dirNameContainsKeyword {
+			// not in quarantine dir
+			continue
+		}
+
+		q.quarantineCounts[dirPath]++
```

Comment:
Do we want to list the names of the quarantined files? It looks like we're only doing count per directory, but maybe I'm missing it?

Reply:
I started off with that but learned that AVs tend to renamed the files to GUIDs then keep some metadata some where. I don't know if they all do this, only learned from seph that crowdstrike does this. Windows defender does the same.

Reply:
After further discussion, re adding file names, even if their just guid's it wont hurt.

Reply:
I'm fine with this, but I'll note that the test host I found had ~600 of them